sycl/include/CL/sycl/backend.hpp

```diff
@@ -121,6 +121,9 @@ make_kernel_bundle(pi_native_handle NativeHandle, const context &TargetContext,
 __SYCL_EXPORT std::shared_ptr<detail::kernel_bundle_impl>
 make_kernel_bundle(pi_native_handle NativeHandle, const context &TargetContext,
                    bool KeepOwnership, bundle_state State, backend Backend);
+__SYCL_EXPORT detail::pi::PiMem make_pi_mem(pi_native_handle NativeHandle,
+                                            context Context, bool KeepOwnership,
+                                            backend Backend);
```

Comment:
why is this pi-based unlike other functions above?

Reply:
This function needs to reside in sycl.so because its implementation needs access to things like context_impl::getHandleRef which is internal and is not exported from sycl.so. I can't implement make_buffer instead of it because it will need to return templated type buffer<T, Dimensions, AllocatorT>. Other make_* functions don't return a templated type, so they are good.
I don't know a nicer way to do this.

Reply:
I see. This is probably the first case where PI objects would cross sycl.so boundaries (of course not counting plugins). Not sure this is a great idea as it creates extra compatibility issues (apps built with earlier versions of PI may not work with a newer PI), but I don't have a better idea at the moment. Please add a comment explaining the decision.

Reply:
I was able to remove this function, so pi_buffer is not exposed now.